Sources/Sampler/abstract_sampler.hpp

```diff
@@ -61,12 +59,19 @@ class AbstractSampler {
     machine_func_ = std::move(machine_func);
   }
 
+  std::shared_ptr<const AbstractHilbert> GetHilbert() const noexcept {
+    return hilbert_;
+  }
+
   const MachineFunction& GetMachineFunc() const noexcept {
     return machine_func_;
   }
 
  protected:
-  AbstractSampler() {
+  std::shared_ptr<const AbstractHilbert> hilbert_;
```

Comment:
I guess we should also set `machine_` to be a protected member here, and initialize it in the constructor of the abstract class, as done with hilbert?

Reply:
Yes, good idea.